src/WeatherRequest.jsx

```diff
@@ -0,0 +1,63 @@
+import { useEffect } from 'react';
+
+const currentWeatherURI = 'https://api.openweathermap.org/data/2.5/weather';
+const hourlyWeatherURI =
+  'https://pro.openweathermap.org/data/2.5/forecast/hourly';
+const dailyWeatherURI =
+  'https://api.openweathermap.org/data/2.5/forecast/daily';
+
+const requestLonLat = async function requestWeatherByLonAndLat(URIparams, URI) {
+  const response = await fetch(
+    `${URI}?${new URLSearchParams(URIparams).toString()}`
+  );
+  const responseJSON = await response.json();
+  return responseJSON;
+};
+
+export default function WeatherRequest(props) {
+  const {
+    coordinates,
+    setCurrentWeather,
+    setHourlyWeather,
+    setDailyWeather,
+    API_KEY,
+  } = props;
+
+  useEffect(() => {
```

Comment:
This useEffect function will run when the component is mounted and every single time one of these value changes,
  ```
  coordinates.longitude,
    API_KEY,
    setCurrentWeather,
    setHourlyWeather,
    setDailyWeather,
```
I'll have too look into it more cause the logic may be fully correct but I will come back to it. It may be better to discuss on call as well!

Reply:
Refactored `useEffect` call and minimized dependencies to just coordinate variables